### PR TITLE
[WIP] add swagger user-related endpoints

### PR DESCRIPTION
**Pull request by jmaunon**
## Description
This is the first PR that includes `swagger` documentation to a better understanding o the endpoints

Endpoints: 

- [ ]  `users/me`
- [x] `users/create_user`
- [ ] `users/login`
- [ ]  `users/`
- [ ] `users/<user_id>`
- [ ] `users/<user_id>/email`
- [ ] `users/<user_id>/role`
- [ ] `users/<user_id>/password`
- [ ] `users/<user_id>/groups`
- [ ] `users/<user_id> (delete)` 
- [ ] `users/search`

## Affected Dependencies
`flasgger`

## How has this been tested?
See `http://localhost:5000/apidocs`

image

- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by jmaunon** [comment]
@victorperezc @IonesioJunior. 
Before continuing with the rest of the endpoints (related with user) I would like to know your feedback to double check

If you like, I would complete with the rest of endpoints and I would include a Note in the main Readme

File: apps/domain/src/main/routes/users/routes.py [hunk 1]
```diff
@@ -1,37 +1,79 @@
...
+              type: object
+              required:
+                - email
+                - password
+              properties:
+                email:
+                  type: string
+                password:
+                  type: string
+                token:
```
Comment (victorperezc) [inline apps/domain/src/main/routes/users/routes.py:48]: The user creation only requires `username` and `password`, the `token` is needed only when calling `POST /setup`

**Review by victorperezc — APPROVED** [review]
Thank you @jmaunon! This truly looks nice.🚀 I'm happy for us to continue documenting the user endpoints, but I'm worried on the added complexity in the code by joining it with the documentation. Have we explored the option to write and store the swagger documentation separate from the code itself? Nevertheless, since the PyGrid routes are already separated from its logic I believe it's okay to add `swagger` documentation as comments on the `routes` functions.

**Comment by madhavajay** [comment]
@jmaunon @victorperezc Perhaps we can use something to autogenerate them: https://github.com/coding-kitties/flask-swagger-generator ?

**Comment by victorperezc** [comment]
@madhavajay auto generating the swagger docs looks more clean to me.

File: apps/domain/src/main/routes/users/routes.py (hunk 1 above)
Comment (jmaunon) [reply apps/domain/src/main/routes/users/routes.py:48]: I though that was optional (at least that is what I can infer from the code), meaning that if you pass the token you could create some especial users such as adminsitrators. see: https://github.com/OpenMined/PyGrid/blob/0a042991da2e77048068cf1f1bd0ca26d0d0995c/apps/domain/src/main/routes/users/routes.py#L37

**Comment by jmaunon** [comment]
> Thank you @jmaunon! This truly looks nice. I'm happy for us to continue documenting the user endpoints, but I'm worried on the added complexity in the code by joining it with the documentation. Have we explored the option to write and store the swagger documentation separate from the code itself? Nevertheless, since the PyGrid routes are already separated from its logic I believe it's okay to add `swagger` documentation as comments on the `routes` functions.

Yep. I was not completely sure if putting everything in a separate `.yaml` or in the specific endpoint, is a tradeoff

**Comment by jmaunon** [comment]
> @jmaunon @victorperezc Perhaps we can use something to autogenerate them: https://github.com/coding-kitties/flask-swagger-generator ?

Hi @madhavajay @victorperezc . I always prefere autogenerate. There are some frameworks for `flask-restful` library but I did not find anything suitable for plain `flask` . I will play a little bit with the library you comment, it seems quite new because there are no documentation, but it could be enough for us. I will let you know

**Comment by IonesioJunior** [comment]
PyGrid has been moved to PySyft repository.

**Pull request closed without merging**